### PR TITLE
[BAD-170] Support for Spark

### PR DESCRIPTION
**Pull request by pavel-sakun**
Initial version of "Spark submit" job entry

### CODE REVIEW

**Comment by pavel-sakun** [comment]
@mattyb149 Matt, could you please take a look? Packaging is both jar and zip which should be easily picked up by kettle's build process.

**Comment by pavel-sakun** [comment]
@brosander you might be interested as well

**Comment by pavel-sakun** [comment]
@deinspanjer

File: src/main/java/org/pentaho/di/job/entries/spark/JobEntrySparkSubmit.java [hunk 1]
```diff
@@ -0,0 +1,544 @@
...
+import static org.pentaho.di.job.entry.validator.JobEntryValidatorUtils.*;
+
+/**
+ * This job entry submits a JAR to Spark and executes a class. It uses the spark-submit script
+ * to submit a command like this.
+ * /Users/jdixon/Documents/test/spark-1.1.0/bin/spark-submit \
+ *  --class org.pentaho.spark.SparkExecTest \
+ *  --master spark://localhost:7077 \
+ *  /Users/jdixon/Documents/workspace-5.0/pentaho-spark/dist/pentaho-spark-TRUNK-SNAPSHOT.jar
+ *
```
Comment (mattyb149) [inline src/main/java/org/pentaho/di/job/entries/spark/JobEntrySparkSubmit.java]: We should update this example to look more like it was run from a PDI install, versus references to someone's home directory, a workspace folder, etc.

**Comment by mattyb149** [comment]
Needs unit and/or integration tests

**Comment by mattyb149** [comment]
One of those unit tests should utilize LoadSaveTester

**Comment by pavel-sakun** [comment]
@mattyb149 tests added

**Comment by mattyb149** [comment]
Cool thanks!

**Pull request merged**